irony-cdb-json.el

```diff
@@ -77,9 +77,22 @@ directories to project directory."
     (let ((elm (nth pos target-list)))
       (append (list elm) (delete elm target-list)))))
 
+(defun irony-cdb-json--cdb-list ()
+  (mapcar #'cdr irony-cdb-json--project-alist))
+
 (defun irony-cdb-json--choose-cdb ()
-  (let ((choices (mapcar (lambda (x) (cdr x)) irony-cdb-json--project-alist)))
-    (completing-read "Choose Irony CDB: " choices nil 'require-match nil)))
+  "Prompt to select CDB from current project root."
+  (let ((proot (irony-cdb-json--find-best-prefix-path
+                (irony-cdb-json--target-path)
+                (mapcar 'car irony-cdb-json--project-alist)))
+        (cdbs '()))
```

Comment:
I believe cdbs could be "more functionally set" by using a filter operation, e.g: cl-remove-if.